internal/apis/certmanager/types_certificaterequest.go

```diff
@@ -126,6 +126,17 @@ type CertificateRequestSpec struct {
 	// of requested `usages`.
 	IsCA bool
 
+	// Requested basic constraints maxPathLen value. Note that the issuer may choose
+	// to ignore the requested maxPathLen value, just like any other requested attribute.
+	//
+	// NOTE: If the CSR in the `Request` field has a BasicConstraints extension,
+	// it must have the same maxPathLen value as specified here.
+	//
+	// If set, maxPathLen must be a value of `0` or greater.
+	// If unset (`nil`), there is no maximum.
+	// Default value is `nil`.
+	MaxPathLen *int
+
```

Comment:
I believe this relates to the question do we encode the BasicConstraints in the CSR?

My goal is to follow the same process that we followed for the KeyUsages:
1. Have IsCA and MaxPathLen in the CR and Cert
2. Encode BasicConstraints in CSR blob by default and add option to Cert to disable this.

Reply:
2 is still behind a feature gate today.

Reply:
I think my view differs a bit; ideally we want a single source of truth for any given value.

That ship has sailed for IsCA which was duplicated into the CR resource (when it could be encoded in basicConstraints).

`usages` is weird because it's an abstraction over both keyusages and extkeyusages, but that information could still have been put into the extensions in the CSR.

Duration has to appear on the CR resource because it's not generally added to a CSR. That's a special case.

But I don't think we need to make maxpathlen a special case. We can always encode it in a CSR, and then if we need to check the value we can parse the CSR to get the value. I'd vote that maxpathlen should only be supported if encoded into the CSR, which would remove the need to have it on the CR resource